core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala

```diff
@@ -38,4 +38,9 @@ class MockWorkspaceManagerDAO extends WorkspaceManagerDAO {
   override def getDataReference(workspaceId: UUID, referenceId: UUID, accessToken: OAuth2BearerToken): DataReferenceDescription = {
     mockReferenceResponse(referenceId)
   }
+
+  override def deleteDataReference(workspaceId: UUID, referenceId: UUID, accessToken: OAuth2BearerToken): Unit = {
+    references.get(referenceId).getOrElse(throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Not found")))
+    references.-=(referenceId)
```

Comment:
tiny tiny style thing - you could also do `references -= (referenceId)` I think that might be personal preference though - so not really a criticism